frontend/utils/src/uri/index.ts

```diff
@@ -25,7 +25,7 @@ export function buildHttpUri(
   path: string
 ): string {
   const fullPath = makePath(pathname, path)
-  return `${protocol}//${hostname}:${port}/${fullPath}`
+  return `${protocol}//${hostname}${port ? `:${port}` : ""}/${fullPath}`
```

Comment:
I updated this related to the comment [here](https://github.com/streamlit/streamlit/pull/12014/files#r2242374398). The `port` can be empty if it is the default port for the respective protocol, which leads to URLs such as "http://foo.bar:/somepath`